hf_hub.py

```diff
@@ -43,32 +47,62 @@ def main():
     )
     args = parser.parse_args()
 
-    # Tell HuggingFace API we need to push the code for the model config and definition.
-    RADIOConfig.register_for_auto_class()
-    RADIOModel.register_for_auto_class("AutoModel")
-
     # Load the checkpoint and create the model.
     checkpoint = torch.load(args.checkpoint_path, map_location="cpu")
     model_args = checkpoint["args"]
-    radio_config = RADIOConfig(
-        vars(model_args), return_summary=True, return_spatial_features=True
-    )
-    radio_model = RADIOModel(radio_config)
+
+    if args.model == "RADIO":
+        # Tell HuggingFace API we need to push the code for the model config and definition.
+        RADIOConfig.register_for_auto_class()
+        RADIOModel.register_for_auto_class("AutoModel")
+
+        radio_model = RADIOModel(radio_config)
+    elif args.model == "E-RADIO":
+        # Tell HuggingFace API we need to push the code for the model config and definition.
+        ERADIOConfig.register_for_auto_class()
+        ERADIOModel.register_for_auto_class("AutoModel")
+
+        radio_config = ERADIOConfig(
+            vars(model_args), return_summary=True, return_spatial_features=True
+        )
+        radio_model = ERADIOModel(radio_config)
+    else:
+        raise ValueError(f"Unknown model {args.model}")
 
     # Restore the model's state.
     state_dict = checkpoint["state_dict"]
     state_dict = clean_state_dict(state_dict)
-    radio_model.model.load_state_dict(
-        get_prefix_state_dict(state_dict, "base_model."), strict=False
-    )
-    radio_model.radio_model.input_conditioner.load_state_dict(
+
+    # Remove keys in state dict whose size does not match that of the model.
+    checkpoint_state_dict = get_prefix_state_dict(state_dict, "base_model.")
+    model_state_dict = radio_model.model.state_dict()
+    for k, v in model_state_dict.items():
+        if k in checkpoint_state_dict:
+            if v.size() != checkpoint_state_dict[k].size():
+                warning(
+                    f"Removing key {k} from state dict due to shape mismatch: "
+                    f"{v.size()} != {checkpoint_state_dict[k].size()}"
+                )
+            else:
+                model_state_dict[k] = checkpoint_state_dict[k]
+    radio_model.model.load_state_dict(model_state_dict, strict=False)
```

Comment:
Is all of this for E-RADIO?

Reply:
E-RADIO needs this if we want to use a different window size than we used during distillation. I was thinking this might be useful for RADIO too? What do you think? Shall I move this piece of code into the E-RADIO case?

Reply:
If it's about window size, then yeah, let's make it specific to E-RADIO. The way that RADIO does it is slightly different.

Reply:
I've made it specific to E-RADIO in the last commit.